### PR TITLE
feat(ams-ui): react forms validation

### PR DESCRIPTION
**Pull request by aashnaPrakash**
Added a new module - react-hook-form for forms validation.
Added a new component that contains reset and submit buttons and can be used across all forms.
Field name and area input validations.

### CODE REVIEW

File: src/components/common/FieldForm.js [hunk 1]
```diff
@@ -100,79 +107,108 @@ const FieldForm = ({ operation, title, selectedRow, closeHandler, submitButtonTe
...
+            control={control}
+            rules={{
+              required: 'Please enter area',
+              validate: (val) =>
+                Number(val) > 0 ? null : 'Enter a valid number',
+            }}
+          />
+        </FormControl>
+        {/* reset/save/create button */}
+        <FormButtons {...{ reset, defaultValues, submitButtonText }} />
```
Comment (gautamguptabasant) [inline src/components/common/FieldForm.js:202]: Should not need to pass `defaultValues` to the button components.
Should have a callback passed instead, which does the resetting/other actions

File: src/components/common/FieldForm.js [hunk 2]
```diff
@@ -100,79 +107,108 @@ const FieldForm = ({ operation, title, selectedRow, closeHandler, submitButtonTe
...
+                  fieldsErrors.fieldName ? fieldsErrors.fieldName.message : null
+                }
+              />
+            }
+            control={control}
+            rules={{
+              required: 'Please enter a field name',
+              pattern: {
+                value: /^([A-Za-z]|[0-9]|_)+$/,
+                message: 'Enter a valid name',
```
Comment (gautamguptabasant) [inline src/components/common/FieldForm.js:172]: How do we convey what is valid? Any ideas on this?

**Review by gautamguptabasant — APPROVED** [review]
Looks good overall. I am guessing there is no more compact way to include it in every field instead of adding the <Controller>.
Also, need to fix the reset button css. See if we can do with an icon instead of the text button.

File: src/components/common/FieldForm.js (hunk 1 above)
Comment (aashnaPrakash) [reply src/components/common/FieldForm.js:202]: Done

File: src/components/common/FieldForm.js (hunk 2 above)
Comment (aashnaPrakash) [reply src/components/common/FieldForm.js:172]: The message / helper text could be provided like - A name must start with a letter followed by letters, digits or  _.
Or something like this below, a popover and help icon.
image

Comment (aashnaPrakash) [reply src/components/common/FieldForm.js:172]: Let me know whatever suits best.

**Comment by aashnaPrakash** [comment]
> Looks good overall. I am guessing there is no more compact way to include it in every field instead of adding the .
> Also, need to fix the reset button css. See if we can do with an icon instead of the text button.

We can discuss this. Just the reset icon next to the submit button was not looking nice.

**Pull request merged**